src/scene/scene.c

```diff
@@ -623,6 +623,31 @@ static int sceneUpdateTurret(struct Scene* scene, void* object) {
     return 1;
 }
 
+void sceneUpdateCameraZoom(struct Scene* scene) {
+    if (playerIsDead(&scene->player)) {
+        scene->camera.fov = DEFAULT_CAMERA_FOV;
+        scene->portalGun.fov = DEFAULT_PORTALGUN_FOV;
+        return;
+    }
+
+    if (scene->player.zoomTimer >= ZOOM_CAMERA_TIME) {
```

Comment:
Minor: it's a bit easier to work with if the timer counts down rather than up. Then the maximum value doesn't need to be checked against  or used in as many places -- just set to the max value when restarting the timer, and a non-zero value means it's running.